core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -0,0 +1,444 @@
+        ContributionAndReviewStatsService,
+    private readonly userService: UserService) {
+  }
+
+  async ngOnInit(): Promise<void> {
+    this.statsData.translationContribution = {};
+    this.statsData.translationReview = {};
+    const userInfo = await this.userService.getUserInfoAsync();
+    const username = userInfo.getUsername();
+
+    if (username === null) {
+      throw new Error('Cannot fetch username.');
+    }
+    this.username = username;
+    const currentOption = this.options.find(
+      (option) => option.contributionType === this.type);
+    this.selectedContributionType = currentOption?.displayName;
+
+    const userContributionRights =
+      await this.userService.getUserContributionRightsDataAsync();
+
+    if (userContributionRights === null) {
+      throw new Error('Cannot fetch user contribution rights.');
+    }
+    const reviewableLanguageCodes = (
+      userContributionRights.can_review_translation_for_language_codes);
+    this.userCanReviewTranslationSuggestions = (
+      reviewableLanguageCodes.length > 0);
+    this.userCanReviewQuestionSuggestions = (
+      userContributionRights.can_review_questions);
+    this.userCanSuggestQuestions = (
+      userContributionRights.can_suggest_questions);
+
+    if (this.userCanReviewTranslationSuggestions) {
+      this.options.push(this.translationReviewOption);
+    }
+    if (this.userCanSuggestQuestions) {
+      this.options.push(this.questionContributionOption);
+    }
+    if (this.userCanReviewQuestionSuggestions) {
+      this.options.push(this.questionReviewOption);
+    }
+
+    await this.fetchStats();
+  }
+
+  toggleDropdown(): void {
+    this.dropdownShown = !this.dropdownShown;
+  }
+
+  toggleMobileDropdown(): void {
+    this.mobileDropdownShown = !this.mobileDropdownShown;
+  }
+
+  async selectOption(contributionType: string): Promise<void> {
+    this.type = contributionType;
+    const currentOption = this.options.find(
+      (option) => option.contributionType === contributionType);
+    this.selectedContributionType = currentOption?.displayName;
+    this.dropdownShown = false;
+    this.mobileDropdownShown = false;
+  }
+
+  async fetchStats(): Promise<void> {
+    const response = await this.contributionAndReviewStatsService.fetchAllStats(
+      this.username);
+
+    if (response.translation_contribution_stats.length > 0) {
+      response.translation_contribution_stats.map((stat) => {
+        const language = this.languageUtilService.getAudioLanguageDescription(
+          stat.language_code);
+        // This throws "Object is possibly 'undefined'.".
+        // We are checking whether there exists translation contribution stats
+        // data for a given language. If there are no stats data or it is
+        // undefined, we are setting it here. Hence we can ignore the ts check
+        // here. We need to suppress this error because of strict type checking.
+        // @ts-ignore
```

Comment:
I think the compiler is complaining about the `language` object, right? I think we can just add a check making sure `language` is defined rather than adding all these ignores below.

Reply:
Done